driver-core/src/test/java/com/datastax/driver/core/SpeculativeExecutionTest.java

```diff
@@ -0,0 +1,218 @@
+package com.datastax.driver.core;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+import com.codahale.metrics.Counter;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import org.scassandra.Scassandra;
+import org.scassandra.ScassandraFactory;
+import org.scassandra.http.client.PrimingClient;
+import org.scassandra.http.client.PrimingRequest;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.datastax.driver.core.exceptions.SyntaxError;
+import com.datastax.driver.core.policies.ConstantSpeculativeExecutionPolicy;
+import com.datastax.driver.core.policies.DowngradingConsistencyRetryPolicy;
+import com.datastax.driver.core.policies.LoadBalancingPolicy;
+
+import static com.datastax.driver.core.Assertions.assertThat;
+
+public class SpeculativeExecutionTest {
+    CCMBridge ccm = null;
+    Scassandra scassandra = null;
+    Cluster cluster = null;
+    SortableLoadBalancingPolicy loadBalancingPolicy;
+    Session session;
+    PrimingClient mockHost1;
+    Counter speculativeExecutions;
+
+    @BeforeClass(groups = "short")
+    public void setup() {
+        // Start a regular CCM cluster and stop host1
+        String clusterName = "speculative_execution";
+        ccm = CCMBridge.create(clusterName, 3);
+        ccm.stop(1);
+        ccm.waitForDown(1);
+
+        // Put an SCassandra instance in host1's place
+        String address = CCMBridge.ipOfNode(1);
```

Comment:
The following _might_ spare you the use of Scassandra: 

In CCMBridge, add:

```java

    public void pause(int n) {
        logger.info("Pausing: " + IP_PREFIX + n);
        execute("ccm node%d pause", n);
    }

    public void resume(int n) {
        logger.info("Resuming: " + IP_PREFIX + n);
        execute("ccm node%d resume", n);
    }
```

Then in your test:

```java
@Test(groups = "short")
    public void should_start_speculative_execution_if_first_execution_takes_too_long() throws InterruptedException {
        loadBalancingPolicy.order = ImmutableList.of(1, 2, 3);
        long startCount = speculativeExecutions.getCount();
        try {
            ccm.pause(1);
            ResultSet rs = session.execute("select v from foo where k = 1");
            Thread.sleep(2000);
            Row row = rs.one();
            assertThat(row.getString("v")).isEqualTo("value from real host");
            assertThat(speculativeExecutions.getCount()).isEqualTo(startCount + 1);
            assertThat(rs.getExecutionInfo().getQueriedHost()).isEqualTo(TestUtils.findHost(cluster, 2));
        } finally {
            ccm.resume(1);
        }
    }
```

At least, it works on my laptop :)